project2/ViewController.swift

```diff
@@ -131,5 +133,59 @@ class ViewController: UIViewController {
         guard let newHigh = highScoreArray.max() else { return }
         highScore = newHigh
     }
+    
+    func registerLocal () {
+        let center = UNUserNotificationCenter.current()
+        center.requestAuthorization(options: [.badge, .sound, .alert]) { (granted, error) in
+            if granted {
+                print("Request allowed.")
+            } else {
+                print("Request denied.")
+            }
+        }
+    }
+    
+    func scheduleLocal () {
+        registerCategories()
+        
+        let center = UNUserNotificationCenter.current()
+        let content = UNMutableNotificationContent()
+        content.body = "Daily reminder to play GuessTheFlag."
+        content.title = "GuessTheFlag"
+        content.sound = .default
+        content.categoryIdentifier = "reminder"
+        content.userInfo = ["currentHighScore": highScore]
+        
+        var dateCompoments = DateComponents()
+        for i in 1...7 {
+            dateCompoments.weekday = i
+            dateCompoments.hour = 18
+        }
+        let trigger = UNCalendarNotificationTrigger(dateMatching: dateCompoments, repeats: true)
+        let request = UNNotificationRequest(identifier: UUID().uuidString, content: content, trigger: trigger)
+        center.add(request)
+    }
+    
+    func registerCategories() {
+        let center = UNUserNotificationCenter.current()
+        center.delegate = self
+        let play = UNNotificationAction(identifier: "play", title: "Play now", options: .foreground)
+        let category = UNNotificationCategory(identifier: "reminder", actions: [play], intentIdentifiers: [])
```

Comment:
"play" and "reminder" are so-called "magic strings" - they appear somewhere and no one knows why.
Sometimes it's okay, but if you use them in multiple places better practice is to just make a constant string and avoid chances of getting them copy-pasted wrong. All identifiers are usually constants instead of just plain strings.